cli/pkg/kctrl/cmd/package/init/init.go

```diff
@@ -0,0 +1,431 @@
+		return err
+	}
+
+	createStep.pkgMetadata.Name = pkgRefName
+	createStep.pkgMetadata.Spec.DisplayName = strings.Split(pkgRefName, ".")[0]
+
+	shortDesc := createStep.pkgMetadata.Spec.ShortDescription
+	if len(shortDesc) == 0 {
+		createStep.pkgMetadata.Spec.ShortDescription = pkgRefName
+	}
+
+	longDesc := createStep.pkgMetadata.Spec.LongDescription
+	if len(longDesc) == 0 {
+		createStep.pkgMetadata.Spec.LongDescription = pkgRefName
+	}
+
+	createStep.pkg.Spec.RefName = pkgRefName
+	err = createStep.Save()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (createStep *CreateStep) printPackageReferenceNameBlock() {
+	createStep.ui.PrintInformationalText("\nA package Reference name must be a valid DNS subdomain name (https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names) \n - at least three segments separated by a '.', no trailing '.' e.g. samplepackage.corp.com")
+}
+
+func (createStep *CreateStep) PostInteract() error {
+	currentTime := metav1.NewTime(time.Now())
+	createStep.pkgMetadata.ObjectMeta.CreationTimestamp = currentTime
+	createStep.pkg.ObjectMeta.CreationTimestamp = currentTime
+	createStep.pkg.Spec.ReleasedAt = currentTime
+	createStep.pkgBuild.CreationTimestamp = currentTime
+	createStep.pkgInstall.CreationTimestamp = currentTime
+
+	err := createStep.updatePackageInstall()
+	if err != nil {
+		return err
+	}
+
+	err = createStep.updatePackage()
+	if err != nil {
+		return err
+	}
+	err = createStep.Save()
+	if err != nil {
+		return err
+	}
+	//createStep.ui.PrintInformationalText(fmt.Sprintf("Both the files can be accessed from the following location: %s\n", createStep.pkgLocation))
+	createStep.printInformation()
+	createStep.printNextStep()
+	return nil
+}
+
+//TODO should we do like this or the way we are generating AppFromAppBuild.
+func (createStep CreateStep) updatePackageInstall() error {
+	existingPkgInstall := createStep.pkgInstall
+	if existingPkgInstall.ObjectMeta.Annotations == nil {
+		existingPkgInstall.ObjectMeta.Annotations = make(map[string]string)
+		existingPkgInstall.ObjectMeta.Annotations[fetch.LocalFetchAnnotationKey] = "."
+	}
+
+	if len(existingPkgInstall.Name) == 0 {
+		existingPkgInstall.Name = createStep.pkgMetadata.Spec.DisplayName
+	}
+
+	if len(existingPkgInstall.Spec.ServiceAccountName) == 0 {
+		existingPkgInstall.Spec.ServiceAccountName = createStep.pkgMetadata.Spec.DisplayName + "-sa"
+	}
+	if existingPkgInstall.Spec.PackageRef == nil {
+		existingPkgInstall.Spec.PackageRef = &v1alpha12.PackageRef{
+			RefName: createStep.pkg.Spec.RefName,
+		}
+	}
+	// TODO Check whether we should add version constraint as well.
+	if len(existingPkgInstall.Spec.PackageRef.RefName) == 0 {
+		existingPkgInstall.Spec.PackageRef.RefName = createStep.pkg.Spec.RefName
+	}
+	return nil
+}
+
+func (createStep CreateStep) updatePackage() error {
+	existingPkg := createStep.pkg
+
+	// TODO What if user has explicitly given the version anything else from 0.0.0?
```

Comment:
What if user has explicitly given the version anything else from 0.0.0 in case of `pkg init`?

Reply:
i think we should just use that version then.

Reply:
Adopted